aws/resource_aws_api_gateway_documentation_part_test.go

```diff
@@ -0,0 +1,243 @@
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "properties", `{"description":"Terraform Acceptance Test"}`),
+					resource.TestCheckResourceAttrSet("aws_api_gateway_documentation_part.test", "rest_api_id"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccAWSAPIGatewayDocumentationPartResponseHeaderConfig(apiName, uProperties),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayDocumentationPartExists("aws_api_gateway_documentation_part.test", &conf),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "location.#", "1"),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "location.0.type", "RESPONSE_HEADER"),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "location.0.method", "GET"),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "location.0.name", "tfacc"),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "location.0.path", "/terraform-acc-test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "location.0.status_code", "200"),
+					resource.TestCheckResourceAttr("aws_api_gateway_documentation_part.test", "properties", `{"description":"Terraform Acceptance Test Updated"}`),
+					resource.TestCheckResourceAttrSet("aws_api_gateway_documentation_part.test", "rest_api_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAPIGatewayDocumentationPartExists(n string, res *apigateway.DocumentationPart) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No API Gateway Documentation Part ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		req := &apigateway.GetDocumentationPartInput{
+			DocumentationPartId: aws.String(rs.Primary.ID),
+			RestApiId:           aws.String(rs.Primary.Attributes["rest_api_id"]),
+		}
+		docPart, err := conn.GetDocumentationPart(req)
+		if err != nil {
+			return err
+		}
+
+		*res = *docPart
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayDocumentationPartDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_documentation_part" {
+			continue
+		}
+
+		req := &apigateway.GetDocumentationPartInput{
+			DocumentationPartId: aws.String(rs.Primary.ID),
+			RestApiId:           aws.String(rs.Primary.Attributes["rest_api_id"]),
+		}
+		_, err := conn.GetDocumentationPart(req)
+		if err != nil {
+			if isAWSErr(err, "NotFoundException", "") {
+				return nil
+			}
+			return err
+		}
+
+		return fmt.Errorf("API Gateway Documentation Part %s/%s still exists.",
```

Comment:
Should be updated based on ID discussion (or at least made a little clear what the second ID is). 😄